### PR TITLE
Add Unit tests in operator

### PR DESCRIPTION
**Pull request by adiprerepa**
Adds unit tests to some yaml helper functions, trying to cover the huge hole in unit tests in operator.

### CODE REVIEW

File: operator/pkg/util/yaml_test.go [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+			desc: "remove-key-val",
+			inBase: map[string]interface{}{
+				"foo": "bar",
+			},
+			inOverlays: map[string]interface{}{
+				"foo": nil,
+			},
+			expectedOverlay: map[string]interface{}{},
+			expectedErr:     nil,
+		},
```
Comment (morvencao) [inline operator/pkg/util/yaml_test.go:148]: Would you also add case with non-nil error?

Comment (adiprerepa) [reply operator/pkg/util/yaml_test.go:147]: Done

**Review by morvencao — APPROVED** [review]

**Pull request merged**